code/campaignclassic/src/main/java/com/adobe/marketing/mobile/AutoCarouselTemplateNotificationBuilder.java

```diff
@@ -0,0 +1,226 @@
+            Log.trace(
+                    CampaignPushConstants.LOG_TAG,
+                    SELF_TAG,
+                    "Only %d image(s) for the auto carousel notification were downloaded. Building"
+                            + " a basic push notification instead.",
+                    downloadedImageUris.size());
+
+            // use the downloaded image if available
+            if (!StringUtils.isNullOrEmpty(downloadedImageUris.get(0))) {
+                pushTemplate.modifyData(
+                        CampaignPushConstants.PushPayloadKeys.IMAGE_URL,
+                        downloadedImageUris.get(0));
+            }
+            final BasicPushTemplate basicPushTemplate =
+                    new BasicPushTemplate(pushTemplate.getData());
+            return BasicTemplateNotificationBuilder.construct(basicPushTemplate, context);
+        }
+
+        smallLayout.setTextViewText(R.id.notification_title, pushTemplate.getTitle());
+        smallLayout.setTextViewText(R.id.notification_body, pushTemplate.getBody());
+        expandedLayout.setTextViewText(R.id.notification_title, pushTemplate.getTitle());
+        expandedLayout.setTextViewText(
+                R.id.notification_body_expanded, pushTemplate.getExpandedBodyText());
+
+        // get custom color from hex string and set it the notification background
+        final String backgroundColorHex = pushTemplate.getNotificationBackgroundColor();
+        AEPPushNotificationBuilder.setElementColor(
+                smallLayout,
+                R.id.basic_small_layout,
+                "#" + backgroundColorHex,
+                CampaignPushConstants.MethodNames.SET_BACKGROUND_COLOR,
+                CampaignPushConstants.FriendlyViewNames.NOTIFICATION_BACKGROUND);
+        AEPPushNotificationBuilder.setElementColor(
+                expandedLayout,
+                R.id.carousel_container_layout,
+                "#" + backgroundColorHex,
+                CampaignPushConstants.MethodNames.SET_BACKGROUND_COLOR,
+                CampaignPushConstants.FriendlyViewNames.NOTIFICATION_BACKGROUND);
+
+        // get custom color from hex string and set it the notification title
+        final String titleColorHex = pushTemplate.getTitleTextColor();
+        AEPPushNotificationBuilder.setElementColor(
+                smallLayout,
+                R.id.notification_title,
+                "#" + titleColorHex,
+                CampaignPushConstants.MethodNames.SET_TEXT_COLOR,
+                CampaignPushConstants.FriendlyViewNames.NOTIFICATION_TITLE);
+        AEPPushNotificationBuilder.setElementColor(
+                expandedLayout,
+                R.id.notification_title,
+                "#" + titleColorHex,
+                CampaignPushConstants.MethodNames.SET_TEXT_COLOR,
+                CampaignPushConstants.FriendlyViewNames.NOTIFICATION_TITLE);
+
+        // get custom color from hex string and set it the notification body text
```

Comment:
Can you split this logic for setting view colors into a separate private method within this class.